template/build/webpack.prod.conf.js

```diff
@@ -27,30 +28,16 @@ const webpackConfig = merge(baseWebpackConfig, {
   output: {
     path: config.build.assetsRoot,
     filename: utils.assetsPath('js/[name].[chunkhash].js'),
-    chunkFilename: utils.assetsPath('js/[id].[chunkhash].js')
+    chunkFilename: utils.assetsPath('js/[name].[chunkhash].js')
   },
   plugins: [
     // http://vuejs.github.io/vue-loader/en/workflow/production.html
     new webpack.DefinePlugin({
       'process.env': env
     }),
-    new UglifyJsPlugin({
-      uglifyOptions: {
-        compress: {
-          warnings: false
-        }
-      },
-      sourceMap: config.build.productionSourceMap,
-      parallel: true
-    }),
     // extract css into its own file
-    new ExtractTextPlugin({
+    new MiniCssExtractPlugin({
       filename: utils.assetsPath('css/[name].[contenthash].css'),
```

Comment:
if you don't provide an explicit chunkFileName here you'll get errors when you split several chunks for lazy loading. (something like trying to emmit to the same file)

Reply:
Thank for looking over the code @gkatsanos!

Are you sure, though? The [readme](https://github.com/webpack-contrib/mini-css-extract-plugin) says both `filename` and `chunkFilename` are optional. And the [webpack docs](https://webpack.js.org/configuration/output/#output-chunkfilename) say:
> By default `[id].js` is used or a value inferred from `output.filename` (`[name]` is replaced with `[id]` or `[id].` is prepended)
I'm assuming it's inferred from `filename` in this case, but haven't verified.

If it does break that way, how would I repro this effectively? I'd probably need more than just a long CSS file to test, right?

In any case, I was trying to mirror the configuration used for the `extract-text-webpack-plugin`, and this combination seemed to produce correct results (eyeballed). It's possible we need to reconsider the default settings.

Reply:
Try having 2 or 3 lazy loaded components using vue-router when each one's using a separate CSS file.

Reply:
@gkatsanos Finally found some time to take another look at this. I have to say, you sent me down the rabbit hole with this one...

First of all, I was able to determine relatively quickly that if `filename` is specified and `chunkFilename` *is not*, webpack uses `filename` by default.

Second, I'm pretty sure we don't need `[contenthash]`, which was used by the `extract-text-webpack-plugin`. The `[chunkhash]` is probably more suitable for our purposes anyway.

However, that's not the end of the story. Turns out that regardless of the naming strategy used here, the current configuration may generate a new `chunkhash` for the `vendor` chunk for any change in the code, even if it is not in `node_modules`. **This is probably a blocker.**

I did a bit of research and played around with different variations of `filename` and `chunkFilename` for both `output` and the `MiniCssExtractPlugin` using different combinations of placeholders (`[name]`, `[id]`, `[contenthash]` and `[chunkhash]`). I also tested with and without explicitly named chunks in dynamic imports for routing (e.g., `const SampleComponent = () => import(/* webpackChunkName: "sample" */ '@/components/SampleComponent')`, which imports a `SampleComponent.vue` file). And I fiddled quite a bit with the `optimization` settings.

**The bottom line is that I've solved it.** I have to clean it up a bit, so I hope to push it in the next day or two.